### PR TITLE
googletest, fix recipe for secondaryArchSuffix

### PR DESCRIPTION
**Pull request by Begasus**
Patch based on their own upstream patch ... https://github.com/google/googletest/pull/1160

### CODE REVIEW

**Comment by Begasus** [comment]
Need to test for x86_64

File: dev-libs/gtest/patches/gtest-1.8.0.patchset [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -4,10 +4,22 @@ project( googletest-distribution )
+ 
+ enable_testing()
+ 
++include(CMakeDependentOption)
++if (CMAKE_VERSION VERSION_LESS 2.8.5)
++  set(CMAKE_INSTALL_BINDIR "bin" CACHE STRING "User executables (bin)")
++  set(CMAKE_INSTALL_LIBDIR "lib${LIB_SUFFIX}" CACHE STRING "Object code libraries (lib)")
```
Comment (fbrosson) [inline dev-libs/gtest/patches/gtest-1.8.0.patchset]: This might work, but you are hardcoding, and this is not recommended.

BTW, I think @mmlr will confirm that `$relativeLibDir` should not be hardcoded in a patch but, instead, passed from the recipe.

Comment (Begasus) [reply dev-libs/gtest/patches/gtest-1.8.0.patchset]: Using -DCMAKE_INSTALL_LIBDIR=$libDir" would be better to use in the recipe then?

Comment (fbrosson) [reply dev-libs/gtest/patches/gtest-1.8.0.patchset]: Yep, I think so. BTW, `dev-qt/prison/` does so:
```
		-DCMAKE_INSTALL_PREFIX:PATH="$prefix" \
		-DCMAKE_INSTALL_LIBDIR:PATH="$libDir" \
		-DCMAKE_INSTALL_INCLUDEDIR:PATH="$includeDir"
```

Comment (Begasus) [reply dev-libs/gtest/patches/gtest-1.8.0.patchset]: Yeah, they probably work because they are mentioned in the cmake file(s), without the GnuInstallDirs it won't work, need to investigate ... :)

Comment (Begasus) [reply dev-libs/gtest/patches/gtest-1.8.0.patchset]: One question (well maybe) :) should I keep the patch and use -DCMAKE_INSTALL_* for the others in the recipe?

Comment (fbrosson) [reply dev-libs/gtest/patches/gtest-1.8.0.patchset]: IDK. I'd have to look closer...

Comment (alaviss) [reply dev-libs/gtest/patches/gtest-1.8.0.patchset]: @Begasus 
Don't forget about [`$cmakeDirArgs`][0]

[0]: https://github.com/haikuports/haikuports/issues/2074

**Comment by Begasus** [comment]
Bumped to git revision (with latest changes), seems to work ok but still getting a problem with pkg-config afterwords, ps not related to the build ... 
https://github.com/google/googletest/issues/1438

File: dev-libs/gtest/gtest-1.9.0~git.recipe [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python3
```
Comment (fbrosson) [inline dev-libs/gtest/gtest-1.9.0~git.recipe]: Python3 has support for secondary architectures, so I guess it would be safer to use it here.
```diff
-	cmd:python3
+	cmd:python3$secondaryArchSuffix
```

Comment (Begasus) [reply dev-libs/gtest/gtest-1.9.0~git.recipe]: @ your service :)

Comment (Begasus) [reply dev-libs/gtest/gtest-1.9.0~git.recipe]: Need to revert this, looks like python3_x86 doesn't provide the cmd:python3_x86 :)

Comment (fbrosson) [reply dev-libs/gtest/gtest-1.9.0~git.recipe]: Oops, my mistake! I'm so sorry!

Well, if, on x86_gcc2, haikuporter activates python3 instead of python3_x86, then it might be worth trying to "convince" it to activate python3_x86.
Maybe adding `devel:libpython3.6m$secondaryArchSuffix` to `BUILD_REQUIRES` will do the trick. Good luck :)

Comment (pulkomandy) [reply dev-libs/gtest/gtest-1.9.0~git.recipe]: Beware, python3_x86 seems to have problems with library paths and may crash.

Comment (Begasus) [reply dev-libs/gtest/gtest-1.9.0~git.recipe]: Should we use python2 here instead then? I don't see any errors on x86_gcc2 (x86) nor x8_64

File: dev-libs/gtest/patches/gtest-1.8.0.patchset (hunk 1 above)
Comment (Begasus) [reply dev-libs/gtest/patches/gtest-1.8.0.patchset]: @alaviss is this needed? we are way past version 2.8.5 already?
Taking a quick look I can't find any recipe forcing a version for cmake (could be that I missed) :)

File: dev-libs/gtest/gtest-1.9.0~git.recipe [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+* Test discovery
+* A rich set of assertions
+* User-defined assertions
+* Death tests
+* Fatal and non-fatal failures
+* Value-parameterized tests
+* Type-parameterized tests
+* Various options for running the tests
+* XML test report generation"
+HOMEPAGE="https://github.com/google/googletest/"
```
Comment (fbrosson) [inline dev-libs/gtest/gtest-1.9.0~git.recipe]:
```diff
-HOMEPAGE="https://github.com/google/googletest/"
+HOMEPAGE="https://github.com/google/googletest"
```

File: dev-libs/gtest/gtest-1.9.0~git.recipe [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python2
```
Comment (fbrosson) [inline dev-libs/gtest/gtest-1.9.0~git.recipe]: @Begasus, in case you would like to revert to python3:
```diff
-	cmd:python2
+	cmd:python3
```
I think this would work, thanks to commit 298e4d49689975287246a3bf99592a169f8f29d8 by @pulkomandy.

Comment (Begasus) [reply dev-libs/gtest/gtest-1.9.0~git.recipe]: Ok, I reverted it since there wasn't any response on whether to use 2 or 3 ...

File: dev-libs/gtest/gtest-1.9.0~git.recipe [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+* A rich set of assertions
+* User-defined assertions
+* Death tests
+* Fatal and non-fatal failures
+* Value-parameterized tests
+* Type-parameterized tests
+* Various options for running the tests
+* XML test report generation"
+HOMEPAGE="https://github.com/google/googletest"
+COPYRIGHT="2008-2016 Google Test developers"
```
Comment (fbrosson) [inline dev-libs/gtest/gtest-1.9.0~git.recipe]: s/201**6**/201**8**/

**Comment by fbrosson** [comment]
FWIW, I have built **gtest** `1.9.0~git` on all possible architectures.
I have also built protobuf_x86 3.5.1.1 on x86_gcc2 using gtest_x86 `1.8.0` vs `1.9.0~git` and the resulting packages are identical.

(Of course, protobuf's static libs have different checksums at every build, so we can ignore this in the comparison. The metadata is also slightly diferent, but that's because the `.PackageInfo` file keeps track of which version of `lib:libgmock` is required by protobuf's base package.)

In other words, I think this PR is ready for a merge.
(Is everyone OK with that?)
